test/test_jit.py

```diff
@@ -6251,12 +6251,60 @@ def forward(self, x):
                     x = torch.neg(x)
                 return x
 
+    def test_onnx_export_script_module_overload(self):
+        class ModuleToExport(torch.jit.ScriptModule):
+            def __init__(self):
+                super(ModuleToExport, self).__init__()
+
+            @torch.jit.script_method
+            def forward(self, x):
+                return (torch.prod(x) +
+                        torch.prod(x, 0) +
+                        torch.prod(x, 0, False) +
+                        torch.sum(x) +
+                        torch.sum(x, 0, False) +
+                        torch.mean(x) +
+                        torch.mean(x, 0) +
+                        torch.mean(x, 0, False))
+
         mte = ModuleToExport()
-        outputs = mte(torch.zeros(1, 2, 3, dtype=torch.long))
+        outputs = mte(torch.zeros(1, 2, 3))
         self.assertExpected(torch.onnx.export_to_pretty_string(
             mte, (torch.zeros(1, 2, 3),), None, verbose=False,
             example_outputs=outputs))
 
+    def test_onnx_export_script_module_overload_fail(self):
+        class ModuleToExport(torch.jit.ScriptModule):
+            def __init__(self):
+                super(ModuleToExport, self).__init__()
+
+            @torch.jit.script_method
+            def forward(self, x):
+                return torch.prod(x, dtype=torch.double)
+
+        mte = ModuleToExport()
+        outputs = mte(torch.zeros(1, 2, 3))
+        with self.assertRaisesRegex(RuntimeError, "Couldn't export operator"):
+            torch.onnx.export_to_pretty_string(
+                mte, (torch.zeros(1, 2, 3),), None, verbose=False,
+                example_outputs=outputs)
+
+    def test_onnx_export_script_module_unsupported_optional(self):
+        class ModuleToExport(torch.jit.ScriptModule):
```

Comment:
how is this different than the test above?

Reply:
This one is verifying that `dtype` can't be exported to ONNX, and raises and exception.

The one above is verifying that the overload switch I added to `symbolic.py`, i.e. `torch.prod(x)` matches `prod(x, None)` export but `torch.prod(x, 0)` matches `prod(x, dim, keepdim, None)` export.

Reply:
test_onnx_export_script_module_overload_fail and test_onnx_export_script_module_unsupported_optional are character-for-character identical.

Reply:
Ah, lol. I wanted to test two different things and they ended up being the exact same code. I remember writing `with self.assertRaisesRegex` twice for this PR with some time in-between.

Will remove one of them.